### PR TITLE
Fix for .NET Standard for Windows and VS 2017.3

### PR DESCRIPTION
**Pull request by Jaykul**
First, fix "Windows" to "Windows_NT" so that it will build (on Windows).

Update the solution file with real targets and fix TokensGeneratorLib project GUID
Update .gitignore with the new output folders and vs `.ide` file
Clean up unnecessary using statements where `System.Threading.Tasks` was included.

### CODE REVIEW

File: Gherkin/Gherkin.csproj [hunk 1]
```diff
@@ -1,8 +1,7 @@
 <Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
-    <TargetFrameworks Condition="'$(OS)' == 'Windows'">netstandard1.5;net45</TargetFrameworks>
+    <TargetFrameworks Condition="'$(OS)' == 'Windows_NT'">netstandard1.5;net45</TargetFrameworks>
```
Comment (SabotageAndi) [inline Gherkin/Gherkin.csproj:3]: Ah, that was the reason for the AppVeyor build fails! Thanks for spotting this.

**Comment by SabotageAndi** [comment]
If you take this change https://github.com/cucumber/gherkin-dotnet/pull/4/files#diff-f463ed690c7b12c57c0421b52a4240aaL25 the unit tests should run again through.

**Comment by SabotageAndi** [comment]
@Jaykul Thanks for fixing this. I was a little bit out of ideas why the AppVeyor build was broken. 
Because of the makefiles I develop Gherkin-dotnet mainly on my Mac.

**Comment by aslakhellesoy** [comment]
Hi @Jaykul,

Thanks for your making your first contribution to Cucumber, and welcome to the Cucumber committers team! You can now push directly to this repo and all other repos under the cucumber organization! 🍾

In return for this generous offer we hope you will:

* ✅ Continue to use branches and pull requests. When someone on the core team approves a pull request (yours or someone else's), you're welcome to merge it yourself.
* 💚 Commit to setting a good example by following and upholding our [code of conduct](https://cucumber.io/conduct) in your interactions with other collaborators and users.
* 💬 Join the [community Slack channel](http://cucumber.io/support#slack) to meet the rest of the team and make yourself at home.
* ℹ️ Don't feel obliged to help, just do what you can if you have the time and the energy.
* 🙋 Ask if you need anything. We're looking for feedback about how to make the project more welcoming, so please tell us!

On behalf of the Cucumber core team,
Aslak Hellesøy
Creator of Cucumber

**Pull request merged**